src/core/literals.py

```diff
@@ -31,6 +30,8 @@
 AuthMechanism = Literal["SASL_PLAINTEXT", "SASL_SSL", "SSL"]
 Scope = Literal["INTERNAL", "CLIENT"]
 DebugLevel = Literal["DEBUG", "INFO", "WARNING", "ERROR"]
+Substrate = Literal["vm", "k8s"]
```

Comment:
Yes, I prefer having the substrate as a literal and avoid passing it around on the classes.
On 1: do you also mean moving other objects like `Status(Enum), StatusLevel, Ports...` to `models.py`?

Reply:
1. Yeah I think so. I think that the I think the `Status` types can probably stay now that I think about it, but `Ports` can go to config maybe? I dunno. Maybe it's best to leave them for now, but if you can think of a really obvious place to put them then we can do that

2. Honestly, fair point! I think that the main use for having it in the `__init__` is for when we want to construct it differently in unit-tests. You're right though. Definitely have a `SUBSTRATE="vm"` var in literals, but otherwise I'm not sure.